packages/@aws-cdk/pipelines/README.md

```diff
@@ -62,6 +107,524 @@ class MyApplication extends Stage {
+    buildEnvironment: {
+      computeType: codebuild.ComputeType.LARGE,
+    },
+
+    // Control Elastic Network Interface creation
+    vpc: vpc,
+    subnetSelection: { subnetType: ec2.SubnetType.PRIVATE },
+    securityGroups: [mySecurityGroup],
+
+    // Additional policy statements for the execution role
+    rolePolicy: [
+      new iam.PolicyStatement({ /* ... */ }),
+    ],
+  },
+
+  assetPublishingCodeBuildDefaults: { /* ... */ },
+  selfMutationCodeBuildDefaults: { /* ... */ },
+});
+```
+
+### Building Docker image assets, and file asset bundling using Docker
+
+Docker can be used in 3 different places in the pipeline:
+
+* If you are using Docker image assets in your application stages: Docker will
+  run in the asset publishing projects.
+* If you are using Docker image assets in your stack (for example as
+  images for your CodeBuild projects): Docker will run in the self-mutate project.
+* If you are using Docker to bundle file assets anywhere in your project (for
+  example, if you are using such construct libraries as
+  `@aws-cdk/aws-lambda-nodejs`): Docker will run in the
+  *synth* project.
+
+For the first case, you don't need to do anything special. For the other two cases,
+you need to make sure that **privileged mode** is enabled on the correct CodeBuild
+projects, so that Docker can run correctly.
+
+If you are using asset bundling, use a `CodeBuildStep` as the `synth` step
+and pass `privileged: true`. If you are using Docker assets in the pipeline stack,
+pass `pipelineUsesDockerAssets: true`.
+
+> **Important**: You must turn on the `privileged` and `pipelineUsesDockerAssets` flags,
+> commit and allow the pipeline to self-update *before* adding the actual Docker asset or
+> bundling step.
+
+The following example shows both:
+
+```ts
+new CodePipeline(this, 'Pipeline', {
+  synth: new CodeBuildStep('Synth', {
+    // If you are using a CodeBuildStep explicitly, set the 'cdk.out' directory
+    // to be the synth step's output.
+    primaryOutputDirectory: 'cdk.out',
+
+    buildEnvironment: {
+      // Turn on privileged mode for asset bundling
+      privileged: true,
+
+      // Using an asset image in the pipeline requires turning on
+      // 'pipelineUsesDockerAssets' below.
+      buildImage: LinuxBuildImage.fromAsset(this, 'Image', {
+        directory: './docker-image',
+      })
+    },
+  }),
+
+  // Turn this on if the pipeline stack uses Docker image assets
+  pipelineUsesDockerAssets: true,
+});
+```
+
+You may also need to authenticate to Docker registries to avoid being throttled.
+See the section **Docker registry credentials** below for information on how to do
+that.
+
+### Arbitrary CodePipeline actions
+
+If you want to add a type of CodePipeline action the CDK Pipeline that
+no class exists for yet, you can define your own class that extends
+`Step` and implements `ICodePipelineActionFactory`.
+
+Here's a simple example that adds a manual approval (this example is
```

Comment:
Maybe we should illustrate with something that _does not_ exist. It's a bit misleading :-)